### PR TITLE
[docker] Refactor container to be more developer-friendly

### PR DESCRIPTION
**Pull request by imphil**
Clean up the Docker container image to make it more developer-friendly.

- Ensure that all OpenTitan dependencies are available.
- Provide a convenient way to run commands within the container as
  current local user, ensuring that file permissions of mounted volumes
  match.
- Give the user within the container sudo rights, if it wants to use it
  e.g. to install additional packages.
- Provide a convenient way to load additional configuration into the
  environment, e.g. the path to a license server.
- Generally clean up the Dockerfile and make it consistent.
- Remove some additionally installed packages where we don't know what
  they are used for. We can always add them back if we find out the use
  case.

### CODE REVIEW

File: util/container/Dockerfile [hunk 1]
```diff
@@ -41,68 +42,76 @@ RUN OBS_URL="https://download.opensuse.org/repositories"; \
...
+    && apt-get update \
+    && xargs apt-get install -y </tmp/apt-requirements.txt \
+    && apt-get install -y \
+        sudo \
+        gosu \
         locales \
         locales-all \
-        gnupg2 \
-        libc6-i386 \
-        libtool \
```
Comment (moidx) [inline util/container/Dockerfile:62]: Do we run the software build in CI with this container? At some point this was added to fix some dependencies with some host tools.

File: util/container/Dockerfile [hunk 2]
```diff
@@ -41,68 +42,76 @@ RUN OBS_URL="https://download.opensuse.org/repositories"; \
...
         minicom \
-        screen && \
-    apt-get clean; \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*
+        screen \
+        dc \
+        time \
+    && apt-get clean; \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/* /tmp/apt-requirements.txt
```
Comment (moidx) [inline util/container/Dockerfile:73]: nit: `/tmp/apt-requirements.txt` seems redundant.

**Review by moidx — APPROVED** [review]
Nice set of changes!

File: util/container/Dockerfile (hunk 1 above)
Comment (imphil) [reply util/container/Dockerfile:62]: Even without explicitly installing those packages (here) the build of host and device software works beautifully (it's using the same apt-requirements.txt as we do in CI).

File: util/container/Dockerfile (hunk 2 above)
Comment (imphil) [reply util/container/Dockerfile]: Removed.

File: util/container/Dockerfile [hunk 3]
```diff
@@ -20,12 +20,13 @@ ARG RISCV_TOOLCHAIN_TAR_VERSION
 ARG RUST_VERSION
 
 LABEL version="1.0"
-LABEL description="OpenTitan container for hardware development."
+LABEL description="OpenTitan development container."
 LABEL maintainer="miguelosorio@google.com"
```
[the remark this answers is not in the document]
Comment (imphil) [reply util/container/Dockerfile]: done.

**Comment by imphil** [comment]
Thanks for your review @moidx, changes have been made! :-)

File: util/container/Dockerfile (hunk 1 above)
Comment (imphil) [reply util/container/Dockerfile:62]: Oh and I missed the initial part of your question: no, CI doesn't use this container (but the same list of packages to install).

**Pull request merged**